src/python/zquantum/core/estimator_test.py

```diff
@@ -163,6 +164,79 @@ def test_get_estimated_expectation_values_with_constant(
         assert len(values) == 2
         assert coefficient == value
 
+    def test_get_estimated_expectation_values_optimal_shot_allocation(
+        self, estimator, backend, circuit, operator, epsilon, delta
+    ):
+        # Given
```

Comment:
I think this test doesn't really check whether the shot allocation strategy is working correctly. That's probably due to the use of `MockBackend` – I would add a `TODO` here, or a story to the backlog, to add more robust tests once the we replace `MockBackend` with `SimpleBackend`.

Reply:
Yup, added a todo for the two optimal shot allocation tests in 1913685791dc9c11d72909115a85591e9c9d51a6.